File: Scala/src/main/scala/org/revian/MyDeque.scala
package org.revian

class MyDeque {
  private var deque = List[Int]()

  def addFirst(value: Int): Unit = {
    deque = value :: deque
  }

  def addLast(value: Int): Unit = {
    deque = deque :+ value
  }

  def removeFirst(): Int = {
    if (deque.isEmpty) throw new NoSuchElementException("Deque is empty")
    val first = deque.head
    deque = deque.tail
    first
  }

  def removeLast(): Int = {
    if (deque.isEmpty) throw new NoSuchElementException("Deque is empty")
    val last = deque.last
    deque = deque.init
    last
  }

  def size(): Int = deque.length
}

File: Java/src/main/java/org/revian/MyDeque.java
package org.revian;

import java.util.LinkedList;

public class MyDeque {
    private LinkedList<Integer> deque;

    public MyDeque() {
        deque = new LinkedList<>();
    }

    public void addFirst(int value) {
        deque.addFirst(value);
    }

    public void addLast(int value) {
        deque.addLast(value);
    }

    public int removeFirst() {
        if (deque.isEmpty()) {
            throw new IllegalStateException("Deque is empty");
        }
        return deque.removeFirst();
    }

    public int removeLast() {
        if (deque.isEmpty()) {
            throw new IllegalStateException("Deque is empty");
        }
        return deque.removeLast();
    }

    public int size() {
        return deque.size();
    }
}
